### PR TITLE
Added scalar object

### PR DESCRIPTION
**Pull request by N-thony**
@rdstern this replaces #9006. Have a look

### CODE REVIEW

**Review by rdstern — DISMISSED** [review]
@N-thony wow, great.  Looks fine now.  I would really like to get this added and also document it.  @Patowhiz over to you.  If small changes in the code are needed, and you are happy to make them, then I would accept that, so it can be merged quickly.

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -2175,11 +2217,11 @@ DataSheet$set("public", "get_object", function(object_name) {
 )
 
 DataSheet$set("public", "rename_object", function(object_name, new_name, object_type = "object") {
-  if(!object_type %in% c("object", "filter", "calculation", "graph", "table","model","structure","summary", "column_selection")) stop(object_type, " must be either object (graph, table or model), filter, column_selection or a calculation.")
-
+  if(!object_type %in% c("object", "filter", "calculation", "graph", "table","model","structure","summary", "column_selection", "scalar")) stop(object_type, " must be either object (graph, table or model), filter, column_selection or a calculation.")
```
Comment (Patowhiz) [inline instat/static/InstatObject/R/data_object_R6.R:2220]:
```suggestion
  if(!object_type %in% c("object", "filter", "calculation", "graph", "table","model","structure","summary", "column_selection", "scalar")) stop(object_type, " must be either object (graph, table or model), filter, column selection, calculation or scalar.")

```

File: instat/static/InstatObject/R/data_object_R6.R [hunk 2]
```diff
@@ -2202,31 +2244,40 @@ DataSheet$set("public", "rename_object", function(object_name, new_name, object_
...
+    names(private$scalars)[names(private$scalars) == object_name] <- new_name
+    self$append_to_metadata(scalar, private$scalars)
+  }
 }
 )
 
 DataSheet$set("public", "delete_objects", function(data_name, object_names, object_type = "object") {
-  if(!object_type %in% c("object", "graph", "table","model","structure","summary","filter", "calculation", "column_selection")) stop(object_type, " must be either object (graph, table or model), filter, column selection or a calculation.")
-
+  if(!object_type %in% c("object", "graph", "table","model","structure","summary","filter", "calculation", "column_selection", "scalar")) stop(object_type, " must be either object (graph, table or model), filter, column selection or a calculation.")
```
Comment (Patowhiz) [inline instat/static/InstatObject/R/data_object_R6.R:2257]:
```suggestion
  if(!object_type %in% c("object", "graph", "table","model","structure","summary","filter", "calculation", "column_selection", "scalar")) stop(object_type, " must be either object (graph, table or model), filter, column selection,  calculation or scala.")

```

File: instat/static/InstatObject/R/instat_object_R6.R [hunk 3]
```diff
@@ -402,8 +412,15 @@ DataBook$set("public", "get_combined_metadata", function(convert_to_character =
...
-      if(length(templist[[j]]) > 1 || is.list(templist[[j]])) templist[[j]] <- paste(as.character(templist[[j]]), collapse = ",")
+    for (j in seq_along(templist)) {
+      if (is.list(templist[[j]]) || length(templist[[j]]) > 1) {
+        if (length(templist[[j]]) > 0) {
+          templist[[j]] <-
+            paste(names(templist[[j]]), " = ", templist[[j]], collapse = ", ")
+        } else {
+          next
+        }
+      }
```
Comment (Patowhiz) [inline instat/static/InstatObject/R/instat_object_R6.R:423]: @N-thony are these changes required for this pull request? I couldn't understand how they relate to the scalars addition.

**Review by Patowhiz — COMMENTED** [review]
@N-thony this looks good. Just a few commit suggestions and a question. Then I'll merge this.

File: instat/static/InstatObject/R/instat_object_R6.R (hunk 3 above)
Comment (Patowhiz) [reply instat/static/InstatObject/R/instat_object_R6.R:423]: @N-thony kindly expound more. I'm not quite sure, what you mean by more general.
1. Does it relate to how @rdstern wants all columns metadata to be displayed? 
2. Is there a comment that he wrote which you could point me to? or a sample screenshot of current and new implementation will be useful.
3. Is this change required for the scalar feature? 

If the change is not needed for the scalar feature and it affects other parts of the code base , I suggest we make this addition in a separate pull request, then we can test for any regression. If it has to be part of this pull request, because the scalar feature relies on it then I'm hope you and @rdstern have tested for regression in the potential parts that rely on this function.

**Comment by rdstern** [comment]
@N-thony and @Patowhiz I had not realised that was possible, but I have now tried it and I like it!

image

The mode is a troblesome summary, because it may not be unique!  So for the yields there are 2 values and here they are.  Wow - let's keep it, if I have understood correctly and this is the code you have been discussing!

**Comment by Patowhiz** [comment]
@rdstern, the feature you requested prompted @N-thony to make changes to a function used in other features related to column metadata. We are currently discussing whether the new presentation changes made by @N-thony are valid for those parts as well. I understand that you find them valid for Scalars, but do you also find them valid in other metadata areas that use the function?

@N-thony, since you have identified the areas that require changes, could you help by checking where else the function you amended is being used? This would guide @rdstern and me on the areas we should test. I found this approach useful to @rdstern and saves everyone's time.

**Review by Patowhiz — APPROVED** [review]
@N-thony thanks for looking into this, I'll close this, noting also a general change that applies to data frame metadata only has been done.

**Comment by Patowhiz** [comment]
@rdstern I've approved this with a note as explained by @N-thony. I'll merge as soon as you approve.

**Review by rdstern — APPROVED** [review]
